src/Pdp/PublicSuffixListManager.php

```diff
@@ -69,18 +70,28 @@ public function refreshPublicSuffixList()
         $publicSuffixListArray = $this->parseListToArray(
             $this->cacheDir . '/' . self::PDP_PSL_TEXT_FILE
         );
-        $this->writePhpCache($publicSuffixListArray);
+
+        // do not empty existing PHP cache file if source TXT is empty
+        if (is_array($publicSuffixListArray) && !empty($publicSuffixListArray)) {
```

Comment:
Or, when it run within `getList()` method (when there's no PHP cache file in some reason), it will stuck all your app